src/nn/activations.rs

```diff
@@ -55,9 +60,99 @@ where
     }
 }
 
+/// Calls [prelu()] with constant value.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct LeakyReLU<E: Dtype>(E);
+
+impl<E: Dtype> ZeroSizedModule for LeakyReLU<E> {}
+impl<E: Dtype> NonMutableModule for LeakyReLU<E> {}
+
+impl<Ax: Axes, S, E: Dtype, D: Device<E>, T: Tape<E, D>> Module<Tensor<S, E, D, T>> for LeakyReLU<E>
+where
+    S: Shape<LastAxis = Ax> + ReduceShape<Ax>,
+    D: PReLUKernel<Tensor<S, E, D>, Tensor<(), E, D>, Output = Tensor<S, E, D>, Elem = E>,
+{
+    type Output = Tensor<S, E, D, T>;
+    type Error = D::Err;
+
+    fn try_forward(&self, input: Tensor<S, E, D, T>) -> Result<Self::Output, D::Err> {
+        let v = D::default().tensor_from_vec(vec![self.0], ());
+        input.try_prelu(v)
+    }
+}
+
+pub mod builder {
+    #[derive(Debug, Copy, Clone, Eq, PartialEq)]
+    pub struct PReLU;
+}
+
+impl<E: Dtype, D: Device<E>> BuildOnDevice<D, E> for builder::PReLU
+where
+    PReLU<E, D>: BuildModule<D, E>,
+{
+    type Built = PReLU<E, D>;
+    fn try_build_on_device(device: &D) -> Result<Self::Built, <D>::Err> {
+        Self::Built::try_build(device)
+    }
+}
+
+/// Calls [prelu()] with learnable value.
+#[derive(Debug, Clone)]
+pub struct PReLU<E: Dtype, D: Device<E>> {
+    a: Tensor<(), E, D>,
+}
+
+impl<E: Dtype, D: Device<E>> Default for PReLU<E, D> {
+    fn default() -> Self {
+        let dev = D::default();
+        Self {
+            a: dev.tensor(E::from_f32(0.05).unwrap()),
+        }
+    }
+}
+
+impl<E: Dtype, D: Device<E>> From<E> for PReLU<E, D> {
+    fn from(value: E) -> Self {
+        let dev = D::default();
+        Self {
+            a: dev.tensor(value),
+        }
+    }
+}
+
+impl<E: Dtype, D: Device<E>> NonMutableModule for PReLU<E, D> {}
+
+impl<Ax: Axes, S, E: Dtype, D: Device<E>, T: Tape<E, D>> Module<Tensor<S, E, D, T>> for PReLU<E, D>
+where
+    S: Shape<LastAxis = Ax> + ReduceShape<Ax>,
+    D: PReLUKernel<Tensor<S, E, D>, Tensor<(), E, D>, Output = Tensor<S, E, D>, Elem = E>,
+{
+    type Output = Tensor<S, E, D, T>;
+    type Error = D::Err;
+
+    fn try_forward(&self, input: Tensor<S, E, D, T>) -> Result<Self::Output, D::Err> {
+        input.try_prelu(self.a.clone())
```

Comment:
Until we have operator fusion, we should avoid doing this, because of the added memory use and overhead this brings.

Reply:
TBH I'd prefer simpler code that can be optimized later with operator fusion. Plus once we add operator fusion we don't have to go back and remove all this extra stuff. The less specialized kernels we have to write the better IMO. I think PReLU is niche enough where its not necessarily on the hot path.

Reply:
I'm not really convinced that the short term performance loss is worth it being convenient for a long-term optimization, but if we are going to do it for this operator, here's a more efficient implementation:
```rust
let scale = self.a.retaped::<T>().broadcast_like(t.shape());
let scaled = input.with_empty_tape() * scale;
(input.scalar_lt(0.0)).choose(scaled, input) 
```

Reply:
I'd like to start doing this more and more because:
1. It lowers the bar to contributing tensor ops because you don't have to be a cuda expert or even write multiple kernels
    1. As we look to adding additional devices (e.g. #597), each additional tensor op with custom kernel makes this harder.
2. Its more maintainable
3. It gives us a wider set of test cases for #607, which is becoming higher priority in my mind as I've look at optimizations recently